pandas/tests/frame/test_api.py

```diff
@@ -363,15 +363,13 @@ def test_transpose(self, float_frame):
         for col, s in mixed_T.items():
             assert s.dtype == np.object_
 
-    def test_swapaxes(self):
+    def test_swapaxes_deprecated(self):
         df = self.klass(np.random.randn(10, 5))
-        self._assert_frame_equal(df.T, df.swapaxes(0, 1))
-        self._assert_frame_equal(df.T, df.swapaxes(1, 0))
-        self._assert_frame_equal(df, df.swapaxes(0, 0))
-        msg = ("No axis named 2 for object type"
-               r" <class 'pandas.core(.sparse)?.frame.(Sparse)?DataFrame'>")
-        with pytest.raises(ValueError, match=msg):
-            df.swapaxes(2, 5)
+
+        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
```

Comment:
Is there a reason you need the `check_stacklevel=False` ? (it seems a simple first level function, not multiple levels deep)

Reply:
I got an error related to stack level, because something else is also deprecated. I can't exactly remember which function was deprecated, but I just thought this was the easy solution (the exact stack level isn't that important...)

Reply:
>  (the exact stack level isn't that important...)

It is relevant for the user experience. But the stacklevel of 2 should be correct. 
But a different deprecation being raised sounds a bit suspicious. Can you check which one it was?

Reply:
It has something to do with ``SparseDataFrame``:

```
    def test_swapaxes_deprecated(self):
        df = self.klass(np.random.randn(10, 5))

        with tm.assert_produces_warning(FutureWarning):
>           swapped = df.swapaxes(0, 1)

pandas\tests\frame\test_api.py:370:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <contextlib._GeneratorContextManager object at 0x000002453AD9E940>, type = None, value = None, traceback = None

    def __exit__(self, type, value, traceback):
        if type is None:
            try:
>               next(self.gen)
E               AssertionError: Warning not set with correct stacklevel. File where warning is raised: C:\Users\TP\Documents\Python\pandasdev\pandasdev\pandas\core\generic.py != C:\Users\TP\Documents\Python\pandasdev\pandasdev\pandas\tests\frame\test_api.py. Warning message: SparseDataFrame is deprecated and will be removed in a future version.
E               Use a regular DataFrame whose columns are SparseArrays instead.
E
E               See http://pandas.pydata.org/pandas-docs/stable/user_guide/sparse.html#migrating for more.
```
 I'll try to work around it.

Reply:
assert_produces_warning doesn't handle stack level correctly with multiple warnings. I'd recommend just `check_stacklevel=False` like you've done.

Reply:
or don't test it for SparseDataFrame? That's deprecated anyway.

Reply:
Could you explain why the stack level important? Seems not so relevant to me.

Reply:
It's not that important here, as I can see you set it correctly. 

But, in general, we want to test it because it ensures that end-users get a warning with the correction "location" indicated. If the stacklevel is set correctly, the warning message points to where this deprecated feature is being used. If not, it points to some other place, which is not useful or even plain confusing.